src/casync-tool.c

```diff
@@ -398,6 +401,15 @@ static int parse_argv(int argc, char *argv[]) {
                         version();
                         return 0;
 
+                case 'l':
+                        r = set_log_level_from_string(optarg);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse log level \"%s\": %m", optarg);
+
+                        arg_log_level = r;
```

Comment:
I use `r` for two things here: error code, and to return the level that was set. Maybe you don't like that a setter returns the value that was set, in such case I can re-work and do it two steps: `parse_log_level_string(optarg, &arg_log_level)`, then `set_log_level(arg_log_level)`.

Also, you might prefer to be strict with the namespace, and have the functions be named `log_set_level_from_string()` (`log_*` in general).